src/components/AqiGraph/AqiGraph.tsx

```diff
@@ -0,0 +1,311 @@
+            const allData: GraphData = {
+              good: [],
+              moderate: [],
+              sensitive: [],
+              unhealthy: [],
+              veryUnhealthy: [],
+              hazardous: [],
+            };
+
+            // Get the local time from the user's browser
+            const currentDate = new Date();
+            const currentDay = currentDate.getDay();
+            const currentHour =
+              currentDate.getHours() +
+              currentDate.getMinutes() / minutesPerHour;
+            let maxAqi = 0; // Keep track of the highest AQI value to scale the graph
+            for (let i = 0; i < aqiBuffer.length; i++) {
+              // Get the index starting at the oldest reading and wrapping
+              // once we reach the end of the buffer
+              const index = (aqiBufferIndex + i) % aqiBuffer.length;
+              const element = aqiBuffer[index];
+              if (element.timestamp) {
+                const date = element.timestamp.toDate();
+                const hour =
+                  date.getHours() + date.getMinutes() / minutesPerHour;
+                let hoursAgo = currentHour - hour;
+                // If the reading is from yesterday
+                if (date.getDay() !== currentDay) {
+                  hoursAgo = currentHour + (hoursPerDay - hour);
+                }
+                // Limit range to [0,24]
+                hoursAgo = Math.max(Math.min(hoursAgo, hoursPerDay), 0);
+                if (element.aqi > maxAqi) {
+                  maxAqi = element.aqi;
+                }
+                const newElement: GraphElement = {x: hoursAgo, y: element.aqi};
+                // Add element to data
+                if (element.aqi <= good) {
+                  allData.good.push(newElement);
+                } else if (element.aqi < moderate) {
+                  allData.moderate.push(newElement);
+                } else if (element.aqi < sensitiveGroups) {
+                  allData.sensitive.push(newElement);
+                } else if (element.aqi < unhealthy) {
+                  allData.unhealthy.push(newElement);
+                } else if (element.aqi < veryUnhealthy) {
+                  allData.veryUnhealthy.push(newElement);
+                } else {
+                  allData.hazardous.push(newElement);
+                }
+              }
+            }
+            setYAxisLimit(Math.max(maxAqi, defaultYLimit));
+            setData(allData);
+          }
+        }
+      });
+    }
+  }, [sensorDocId]);
+
+  // The graph doesn't include the hazardous category unless thee are data points
+  // that fall in that category. Any time the maximum y-axis value changes, this effect
+  // updates the styling for the graph accordingly
+  useEffect(() => {
+    if (yAxisLimit === defaultYLimit) {
+      // Don't include hazardous color and extra grid line
+      /* eslint-disable-next-line no-magic-numbers */
+      setYAxisTicks([0, 50, 100, 150, 200, 300]);
```

Comment:
I think you could use your variables from above here, the `const good = 50`, etc. Maybe put those at the top of the component?